### PR TITLE
Completed code challenge. Running with Node.js

### PR DESCRIPTION
**Pull request by kevintbradbury**
I opted to use Node.js because this is running from a shell script and would be the most lean solution. Please let me know if you have questions.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+for (var i = 0; i < lines.length; i++){
+    let split = lines[i].split(new RegExp(/=/));
+    json[split[0]] = split[1];
+}
+
+console.log('Template: \n' + templateFile);
+    
+var regexVals = [];
+var keys = ['name','product','gift','gift-value', 'representative'];
```
Comment (spullen) [inline app.js]: The solution should work with any data/template pairings, not just the ones supplied in the test directory (see example in readme). Could you update your solution to handle arbitrary cases as well.

Comment (kevintbradbury) [reply app.js]: Script now obtains keys from template file, looks in data file for equal number of Key/Value pairs.

File: run.sh [hunk 2]
```diff
@@ -4,14 +4,11 @@ if [[ $1 ]]
 then
    if [[ -e "$1" ]]
    then
-       edit-me-to-run-your-program "$1"
+       node app.js "$1" "$2" "$3"
```
Comment (eliottwiener-thirdchannel) [inline run.sh]: `run.sh` should be able to be invoked the way it is invoked in `test.sh`:

https://github.com/kevintbradbury/code-challenge/blob/c243a7f4259663ea5d1fc9b9b45a1f9676b3e018/test.sh#L7

It should work with STDIN and STDOUT, as specified by the `README.md`:

>- Read a data file containing variable bindings from STDIN, parsing it into an in-memory representation.
>...
>- Output the result to STDOUT.
 
Additionally, you should be run the test suite (`./test.sh`) and receive a `Tests passed` result. There are some cases included in the tests that are not handled in your solution. Can you fix them?

File: data [hunk 3]
```diff
@@ -0,0 +1,5 @@
+name =Robert Paulson
+product=ALBOAPPA Chair
+gift =a SVIKTRA Towel Set
```
Comment (eliottwiener-thirdchannel) [inline data:4]: How would your program fare if this line had more whitespace, like:

```
gift                               =a SVIKTRA Towel Set
```

File: data [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+gift-value=$20
```
Comment (eliottwiener-thirdchannel) [inline data]: Try changing the value to `$10` - you might be surprised at the result! See if you can fix the bug.

**Review by eliottwiener-thirdchannel — CHANGES_REQUESTED** [review]
This is looking really good so far! Just needs a few changes to be perfect. Most importantly, see if you can get it to run against the test suite and pass.

**Review by eliottwiener-thirdchannel — APPROVED** [review]

**Pull request closed without merging**